src/components/AssetInfo/TokenSideBarHolder.tsx

```diff
@@ -26,16 +27,17 @@ const isSurrenderDocumentError = (error: any): error is ErrorType =>
 const TokenSideBarHolder = ({
   isEqualBeneficiaryAndHolder,
   approvedBeneficiaryAddress,
+  approvedHolderAddress,
   handleInputChange,
   newHolder,
   transferHoldership,
   changeBeneficiary,
   surrenderDocument,
   error
 }: TokenSideBarHolderProps) => {
-  const showChangeBeneficiary = !!approvedBeneficiaryAddress || isEqualBeneficiaryAndHolder;
-  const isApprovedBeneficiaryAddress = !!approvedBeneficiaryAddress && !isEqualBeneficiaryAndHolder;
-
+  const isApprovedAddress = !!approvedBeneficiaryAddress && !!approvedHolderAddress;
```

Comment:
wasn't this logic extracted out?

Reply:
we extracted out the logic to get the role i.e beneficiary, holder to display the sidebar content and name. this is just to check for the approved addresses and make the input field disable or not based on that.